idaes/power_generation/control/tests/test_pid_pump.py

```diff
@@ -0,0 +1,224 @@
+# Import Pyomo libraries
+import pyomo.environ as pyo
+from pyomo.network import Arc
+
+from idaes.core import FlowsheetBlock
+from idaes.power_generation.unit_models.watertank import WaterTank
+from idaes.power_generation.control.pid_controller import PIDController
+import idaes.core.util.scaling as iscale
+from idaes.power_generation.unit_models.helm import (
+    HelmValve as WaterValve,
+    HelmIsentropicCompressor as WaterPump,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.generic_models.properties import iapws95
+from idaes.core.util.dyn_utils import (
+    copy_values_at_time,
+    copy_non_time_indexed_values)
+
+import pytest
+
+solver_available = pyo.SolverFactory('ipopt').available()
+prop_available = iapws95.iapws95_available()
+
+
+def set_scaling_factors(m):
+    """ Set scaling factors for variables and expressions. These are used for
+    variable scaling and used by the framework to scale constraints.
+
+    Args:
+        m: plant model to set scaling factors for.
+
+    Returns:
+        None
+    """
+    # Set scaling factors for boiler system
+    iscale.set_scaling_factor(
+        m.fs.tank.control_volume.energy_holdup, 1e-10)
+    iscale.set_scaling_factor(
+        m.fs.tank.control_volume.material_holdup, 1e-6)
+    if m.dynamic:
+        for t, c in m.fs.tank.control_volume\
+                .energy_accumulation_disc_eq.items():
+            iscale.constraint_scaling_transform(c, 1e-6)
+
+    # scaling factor for control valves
+    for t in m.fs.config.time:
+        iscale.set_scaling_factor(
+            m.fs.valve.control_volume.properties_in[t].flow_mol, 0.001)
+
+    # Calculate calculated scaling factors
+    iscale.calculate_scaling_factors(m)
+
+
+def main():
```

Comment:
This is a really weird way to write tests.

For one, I would suggest two separate tests for the steady-state and dynamic cases to better identify where the problem is if something breaks.

Secondly, there really should be unit tests in here as well to make sure the models build correctly and look right before you try solving them.

Reply:
we followed the example of the generic pid controller
this is more like a flowsheet example any test ends being an integration test

Reply:
My point is that there really should be a separate set of unit tests of the controller - these do not need to be big for complicated. Jut build a flowsheet, put two `Vars` on it (the measured and controlled variables), add an instance of the controller and check that it has the right set of constraints. do this for dynamic and steady-state and you are done.